### PR TITLE
Clean warnings in examples

### PR DESCRIPTION
**Pull request by jenshnielsen**
These are mostly numpy warnings

* Make sure indices are integers
* The numpy check for positive-semidefinite has changes, so change the data in power_norm_demo.py to not see the warning
* Dont forward trim from axisartist to ticker where it has no effect.

### CODE REVIEW

File: examples/pylab_examples/centered_ticklabels.py [hunk 1]
```diff
@@ -45,6 +45,6 @@
     tick.tick2line.set_markersize(0)
     tick.label1.set_horizontalalignment('center')
 
-imid = len(r)/2
+imid = int(len(r)/2)
```
Comment (jenshnielsen) [inline examples/pylab_examples/centered_ticklabels.py]: Sure, I dont have any strong preference for one over the other

File: examples/pylab_examples/quiver_demo.py [hunk 2]
```diff
@@ -75,8 +75,13 @@
 # 6
 plt.figure()
 M = np.zeros(U.shape, dtype='bool')
-M[U.shape[0]/3:2*U.shape[0]/3,
-  U.shape[1]/3:2*U.shape[1]/3] = True
+XMaskStart = U.shape[0]//3
```
Comment (QuLogic) [inline examples/pylab_examples/quiver_demo.py]: This name is a bit un-Pythonic, but otherwise the PR is fine.

**Comment by tacaswell** [comment]
how far back do we want to backport this?

**Comment by jenshnielsen** [comment]
I probably made a bit of a mess of this 58a27bf4979dd991ee58756b4ff957fd62af9575 does not make sense to backport further than 2.0 (since it relates to a arg deprecated in 2.0) 

Thes rest could be back ported to 1.5.x, The numpy related changes are probably the most serious ones. They will brake code with some future version of numpy.

**Comment by tacaswell** [comment]
Lets just do back to 2.0 then.

**Comment by tacaswell** [comment]
backported to v2.x as c455e52

**Pull request merged**